technical-reports/format/types.md

```diff
@@ -12,21 +12,42 @@ If an explicit type is set, but the value does not match the expected syntax the
 
 ## Color
 
-Represents a 24bit RGB or 24+8bit RGBA color in the sRGB color space. The `$type` property MUST be set to the string `color`. The `$value` property MUST be a string containing a [hex triplet/quartet](https://www.w3.org/TR/css-color-4/#typedef-hex-color) including the preceding `#` character. To support other color spaces, such as HSL, translation tools SHOULD convert color tokens to the equivalent value as needed.
+Represents a color in the UI. The `$type` property MUST be set to the string `color`. The `$value` property MUST be an object string containing a string `colorSpace`, a numeric `coordinates` array, optional numeric `alpha`, string `colorProfile`, and string `fallback` to support legacy color spaces as needed.
+
+| Key            |      Type      | Required | Description                                                                                                                              |
+| :------------- | :------------: | :------: | :--------------------------------------------------------------------------------------------------------------------------------------- |
+| `colorSpace`   |    `string`    |    Y     | An string representing the color space.                                                                                                  |
+| `colorProfile` |    `string`    |          | String representing url of color profile.                                                                                                |
```

Comment:
I think we are not using colorProfile, and just aligning with CSS Color Module 4? `colorSpace` is a shorthand for space + profile + gamut, also borrowing from the same spec

```suggestion
```

Reply:
In [CSS Color Module 4](https://drafts.csswg.org/css-color-4/#terminology) color terminology section, here's the definition of color space:

> A color space is an organization of colors with respect to an underlying colorimetric model, such that there is a clear, objectively-measurable meaning for any color in that color space. This also means that the same color can be expressed in multiple color spaces, or transformed from one color space to another, while still looking the same.

[Color profile](https://drafts.csswg.org/css-color-5/#at-profile) is different, it's about using a particular ICC color profile for device specific implementations (print, for example). For example, you can use the CMYK color space and choose more than one color profile to implement those colors.

Reply:
You’re 100% right, and yes I do see us needing this. As we talked about earlier today, perhaps we limit this initial PR to CSS Color Module 4 where:

- only `colorSpace` is used
- `colorSpace` is limited to the allowed values in CSS Color Module 4 (`srgb`, `srgb-linear`, `display-p3`, etc.)

Then, we open a separate PR to expand the format to include CSS Color Module 5’s concept of `@color-profile`. That way we can get feedback on that separately. Not being prescriptive, but if we copy CSS Color Module 5’s mental model, I could see it introducing a new `$type: 'colorProfile'` token type that maps 1:1 to `@color-profile` (perhaps; maybe we think of another pattern!)